src/handlers/actions/unlockToken.ts

```diff
@@ -14,6 +25,34 @@ export default async (event: ContractEvent): Promise<void> => {
   const colonyClient = await getCachedColonyClient(colonyAddress);
   const tokenAddress = await colonyClient.getToken();
 
+  const { data } =
+    (await query<
```

Comment:
Here, you probably want to include some mechanism to determine if you've hit the implicit limit (100), and continue fetching until you've got them all. 

> For list queries the default number of objects returned is 100. You can override this behavior by setting the limit argument.

https://docs.amplify.aws/cli/graphql/data-modeling/#model-directive


You can imagine a situation where there are 101 colonies using this token as their native token and currently we'll miss that last 1. 

You should be able to follow the same approach we use for pagination, i.e check if there's a nextToken and if so, repeat the query with that next set of responses, building up an array of responses as you go, until there's no nextToken.

And you can increase the limit also to decrease the number of times this loop will need to run. Given that we're not actually paginating anything, I don't see the harm in setting a larger limit (e.g 1000).

Reply:
@ArmandoGraterol Looks good, I would just check the return value of the query function. Internally it accesses the actual data, so I would expect you to be able to write:

```
const {items: colonies, nextToken} = await query()...
```

Reply:
And maybe leave a short comment in the code reminding future us what we're up to here.